tests/test_fileio.py

```diff
@@ -64,14 +66,40 @@ async def test_wrap_file(self, tmp_path: pathlib.Path) -> None:
 
 
 class TestPath:
+    class ChangeDirectory(contextlib.AbstractAsyncContextManager):
+        def __init__(self, cwd: Path) -> None:
+            super().__init__()
+            self.__cwd: Path = cwd
+            self.__pwd: Path | None = None
+
+        async def __aenter__(self) -> Path:
+            self.__pwd = await Path.cwd()
+            await to_thread.run_sync(os.chdir, self.__cwd)
+            return self.__cwd
+
+        async def __aexit__(
+            self,
+            exc_type: type[BaseException] | None,
+            exc_value: BaseException | None,
+            traceback: TracebackType | None,
+        ) -> bool | None:
+            if self.__pwd is not None:
+                await to_thread.run_sync(os.chdir, self.__pwd)
+            return await super().__aexit__(exc_type, exc_value, traceback)
```

Comment:
This seems a bit overkill, why not use `monkeypatch.chdir()` instead?

Reply:
First time I've ever heard of it 😅

That's much simpler. I've scoped it using `monkeypatch.context()` to try to minimize error cases

Reply:
Explain – how does that help?

Reply:
The pytest docs have a warning:

> Mind that patching stdlib functions and some third-party libraries used by pytest might break pytest itself, therefore in those cases it is recommended to use [MonkeyPatch.context()](https://docs.pytest.org/en/latest/reference/reference.html#pytest.MonkeyPatch.context) to limit the patching to the block you want tested

The note is [here](https://docs.pytest.org/en/latest/how-to/monkeypatch.html#global-patch-example-preventing-requests-from-remote-operations). Since pytest is highly dependent on locating and working with tests relative to file paths, I figured `os.chdir` was semi-dangerous to patch. Scoping it like this seems like a good precaution

Reply:
You're not patching `os.chdir()`, you're just switching the working directory.

Reply:
Docs: https://docs.pytest.org/en/6.2.x/reference.html#pytest.MonkeyPatch.chdir

Reply:
My mistake. Fixed. I greatly dislike the naming of that fixture / chdir belongs somewhere else - but that's for me to take up with the pytest folks.